wireapp/wire-android
# feat: unlock app with biometrics (WPB-4696)

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4696" title="WPB-4696" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-4696</a>  [Android] Use Biometric for unlocking the app if available  
  </td></table>
  <br />
 






















----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Description

This PR will allow the user to unlock his device using biometrics (fingerprint or FaceId). 

The biometrics prompt dialog will be displayed if you have your fingerprint or faceId configured in device otherwise regular app lock screen will be shown.

Needs releases with:


### Testing

#### Test Coverage (Optional)


#### How to Test

- Make sure that you have 
- Enable app lock
- Put the app in background for 60s
- Open the app again, you should see biometrics prompt dialog.

### Attachments (Optional)

| Samsung S22 Fingerprint  | Samsung S22 FaceId |
| ------------- | ------------- |
| ![Wire 2023-10-12 at 2_59 PM](https://github.com/wireapp/wire-android-reloaded/assets/18124919/0d136536-ad9f-4699-a4cb-29bf1cdc9b87)  | ![Wire 2023-10-12 at 2_59 PM2](https://github.com/wireapp/wire-android-reloaded/assets/18124919/896a7a30-a4d6-4c30-a487-645e2c2577bf)  |


| Pixel 6 Fingerprint  | 
| ------------- | 
| ![Wire 2023-10-12 at 4_02 PM](https://github.com/wireapp/wire-android-reloaded/assets/18124919/2dccd1da-4a9c-4f48-b093-19079895b39b)  | 



----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
